File: onp/python3/onp.py
# 1. Priority of operators
# 2. No two operators of same priority can stay together in stack colomn
# 3. Lowest priority cannot be placed before highest priority
# 4. ( + ) Pop until find start bracket

t = input()
for t in range(int(t)):
    expression = list(input())
    stack = []
    rpn = ""
    
    for symbol in expression:
        if symbol.isalpha():
            rpn += symbol
        else:
            if symbol == ')':
                found = False
                while not found:
                    stackSymbol = stack.pop()
                    rpn += stackSymbol if stackSymbol != '(' else ''
                    found = stackSymbol == '('
            elif symbol == '^':
                stopPop = False
                while not stopPop:
                    if stack[-1] == '+' or stack[-1] == '-' or stack[-1] == '*' or stack[-1] == '/' or stack[-1] == '^':
                        rpn += stack.pop()
                    else:
                        stack.append(symbol)
                        stopPop = True
            elif symbol == '*' or symbol == '/':
                stopPop = False
                while not stopPop:
                    if stack[-1] == '+' or stack[-1] == '-' or stack[-1] == '*' or stack[-1] == '/':
                        rpn += stack.pop()
                    else:
                        stack.append(symbol)
                        stopPop = True
            elif symbol == '+' or symbol == '-':
                stopPop = False
                while not stopPop:
                    if stack[-1] == '+' or stack[-1] == '-':
                        rpn += stack.pop()
                    else:
                        stack.append(symbol)
                        stopPop = True
            else:
                stack.append(symbol)
    print(rpn)

File: onp/cpp/onp.cpp
#include <iostream>
#include <stack>
#include <ctype.h>
#include <string>
using namespace std;

int main() {
	int testCase = 0;
	cin >> testCase;
	for (int t = 0; t < testCase; t++){
		stack <char> s;
		string rpn = "";
		
		string expression = "";
		cin >> expression;
		
		for (char symbol : expression) {
			if (isalpha(symbol)) {
				rpn += symbol;
			} else {
				if (symbol == ')') {
					bool found = false;
					while (!found) {
						rpn += s.top() != '(' ? s.top() : '\0';
						found = s.top() == '(';
						s.pop();
					}
				} else if (symbol == '^') {
					bool stopPop = false;
					while (!stopPop) {
						if (s.top() == '+' || s.top() == '-' || s.top() == '*' || s.top() == '/' || s.top() == '^') {
							rpn += s.top();
							s.pop();
						} else {
							s.push(symbol);
							stopPop = true;
						}
					}
				} else if (symbol == '*' || symbol == '/') {
					bool stopPop = false;
					while (!stopPop) {
						if (s.top() == '+' || s.top() == '-' || s.top() == '*' || s.top() == '/') {
							rpn += s.top();
							s.pop();
						} else {
							s.push(symbol);
							stopPop = true;
						}
					}
				} else if (symbol == '+' || symbol == '-') {
					bool stopPop = false;
					while (!stopPop) {
						if (s.top() == '+' || s.top() == '-') {
							rpn += s.top();
							s.pop();
						} else {
							s.push(symbol);
							stopPop = true;
						}
					}
				} else {
					s.push(symbol);
				}
			}
		}
        cout << rpn << endl;
    }
	return 0;
}